test/liquid/combinators/lexical_token_test.exs

```diff
@@ -66,14 +77,20 @@ defmodule Liquid.Combinators.LexicalTokenTest do
   end
 
   test "list values" do
-    test_combinator("product[0]", &Parser.value/1, value: {:variable, ["product", "[0]"]})
+    test_combinator("product[0]", &Parser.value/1, value: {:variable, ["product", {:index, [0]}]})
```

Comment:
I thought we were going to parse the numbers as `{:index, 0}`

Reply:
Yes, but it is more consistent (always returns a list) and therefore more easily processed by a recursive function

Reply:
But the index is always a static integer, or a `{:variable, [variable_name]`. So why wrap the static integer in a list and not just return the integer? I don't see how it is more easily processed by a recursive function.